src/d_main.c

```diff
@@ -571,6 +573,74 @@ void D_StartTitle (void)
 // print title for every printed line
 static char title[128];
 
+char **tempdirs = NULL;
+
+static void AutoLoadWADs(const char *path);
+
+static boolean D_AddZipFile(const char *file)
+{
+  int i;
+  mz_zip_archive  zip_archive;
+  char *str, *tempdir;
+  static int idx = 0;
+
+  str = M_StringDuplicate(file);
+  M_ForceLowercase(str);
+
+  if (!M_StringEndsWith(str, ".zip"))
+  {
+    free(str);
+    return false;
+  }
+
+  free(str);
+
+  str = M_StringJoin("_woof_", M_BaseName(file), NULL);
+  tempdir = M_TempFile(str);
+  free(str);
+
+  memset(&zip_archive, 0, sizeof(zip_archive));
+  if (!mz_zip_reader_init_file(&zip_archive, file, MZ_ZIP_FLAG_DO_NOT_SORT_CENTRAL_DIRECTORY))
+  {
+    printf("D_AddZipFile: Failed to open %s\n", file);
+    free(tempdir);
+    return true;
+  }
+
+  M_MakeDirectory(tempdir);
+
+  for (i = 0; i < (int)mz_zip_reader_get_num_files(&zip_archive); ++i)
```

Comment:
Can we evaluate this only once please?

Reply:
It returns constant https://github.com/fabiangreffrath/woof/blob/9aa0dbe1d2b92047cc1322b3eeefb8fec3f4124e/miniz/miniz.c#L7656